tools/make/diplomat-coverage/src/main.rs

```diff
@@ -339,22 +340,51 @@ fn collect_public_types(krate: &str) -> impl Iterator<Item = (Vec<String>, ast::
                     }
                     ItemEnum::Function(_) => {
                         let doc_type = match inside {
-                            Some(In::Enum(tr)) | Some(In::Struct(tr))
+                            Some(In::Enum(tr)) | Some(In::Struct(tr)) | Some(In::Type(tr))
                                 if check_ignored_assoc_item(item_name, tr) =>
                             {
                                 return
                             }
                             Some(In::Enum(_)) => ast::DocType::FnInEnum,
                             Some(In::Trait) => ast::DocType::FnInTrait,
                             Some(In::Struct(_)) => ast::DocType::FnInStruct,
+                            Some(In::Type(_)) => ast::DocType::FnInTypedef,
                             _ => ast::DocType::Fn,
                         };
                         insert_ty(types, path, doc_type);
                     }
                     ItemEnum::Macro(_) => {
                         insert_ty(types, path, ast::DocType::Macro);
                     }
-                    ItemEnum::TypeAlias(_) => {
+                    ItemEnum::TypeAlias(t) => {
```

Comment:
issue: I'm not actually in favor of type aliases being considered items that need coverage.

half the time I want to _ignore_ type aliases and reexports (we have a bunch of both in the allowlist).

When we do this type of thing we are typically in a situation where the type alias is dealing with generics; and that means that the Rust and Diplomat APIs _will_ diverge in some fashion. Having nitpicky checks about that divergence seems counterproductive to me.

The Rust-side choice of which aliases to have is about convenience. The Diplomat-side choice is about exposed functionality. These are not going to lead to the same sets of aliases a lot of the time.

Reply:
Can be convinced otherwise.

Reply:
I think this is a case-by-case determination and we can use the allowlist for cases where we don't want to link all of them.

Reply:
Hm. It would be nice if we had a special allowlist that would disable types from having their aliases considered. Right now you end up having to allowlist everything.

An alternative solution would be to allow rustlinks to use `*` which helps reduce the number of links in such situations.

Reply:
right now I'm allowlisting 4 types, which is manageable